agent/main.go

```diff
@@ -133,51 +134,53 @@ func agentManifestURL(cfg *agentConfig) string {
 }
 
 func updateAgents(cfg *agentConfig, cancel <-chan interface{}) {
-	// Self-update
-	agentPollURL := agentManifestURL(cfg)
-	log.Info("Updating self from ", agentPollURL)
+	if cfg.SelfUpdate {
```

Comment:
nit: `if !cfg.SelfUpdate { return }` would make diff simpler

Reply:
But that is not the intended behaviour - it's still supposed to update all the other stuff if you turn off self-update.

Reply:
The diff looks simpler if you click on 'Hide whitespace changes'